src/util/uploads.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * @flow
+ * @file Utility functions for uploads
+ * @author Box
+ */
+
+/**
+ * Returns true if the given object is a Date instance encoding a valid date
+ * (i.e. new Date('this is not a timestamp') should return false).
+ *
+ * Code adapted from
+ * http://stackoverflow.com/questions/1353684/detecting-an-invalid-date-date-instance-in-javascript
+ *
+ * @param {Date} date
+ * @returns {boolean}
+ */
+function isValidDateObject(date: Date) {
```

Comment:
I would add flow return types to each of the functions function

Reply:
done